Raven.Database/Commercial/ValidateLicense.cs

```diff
@@ -99,10 +99,20 @@ private void ExecuteInternal(InMemoryRavenConfiguration config, bool firstTime =
 				},firstTime:firstTime);
 
 				var attributes = new Dictionary<string, string>(alwaysOnAttributes, StringComparer.OrdinalIgnoreCase);
-				foreach (var licenseAttribute in licenseValidator.LicenseAttributes)
-				{
-					attributes[licenseAttribute.Key] = licenseAttribute.Value;
-				}
+
+			    Monitor.Enter(licenseValidator.LicenseAttributesLock);
```

Comment:
Did you get all the places that read it as well? Otherwise we didn't do much.
Can we create a new instance, isntead?